### ISSUE TITLE
`scope: heading` incorrectly matches body text when paragraph starts with a Markdown link

### ISSUE THREAD

**Issue by theletterf** [jdkato/txtlint#1057]
When a paragraph begins with a Markdown link that contains a file path ending in `.md`, Vale incorrectly applies rules with `scope: heading` to the body text, flagging the period in the `.md` file extension.

## Rule

```yaml
extends: existence
level: warning
nonword: true
message: "Don't end headings with punctuation"
scope: heading
action:
  name: remove
tokens:
  - '[.?!]$'
```

## Input (test.md)

```markdown
# Elastic style guide for Vale

[Vale](https://github.com/errata-ai/vale) is a prose linter that checks documentation against the [Elastic style guide](/contribute-docs/style-guide/index.md). More text here.
```

## Expected behavior

No warning — line 3 is body text, not a heading, so `scope: heading` should not match.

## Actual behavior

```
test.md
 3:178  warning  Don't end headings with punctuation
```

The period in `index.md` (inside the link path) is incorrectly flagged.

## Notes

- The issue only occurs when the paragraph starts with a Markdown link.
- Removing the link at the start of the paragraph eliminates the false positive.
- The link's destination path (`/contribute-docs/style-guide/index.md`) contains the period being matched.

**Comment by jdkato** [comment]
I also can't reproduce this -- it appears the output you've given is from a different file than you've shared, as the line lengths don't match.
